### DISCUSSION TITLE
Limit user permissions to only edit items from defined list of other users

### DISCUSSION THREAD

**Discussion by luciandavidescu** [directus/directus#17459] [Archive]
So a user's capabilities can be limited to see/update/delete only his own set of items, with a filter rule such as `User Created / Equals / $CURRENT_USER` (that would be an "author").

Next level is a user who can see/update/delete any other user's items (that's an "editor").

How about an intermediate level, where a user is allowed to see/update/delete the items of a certain group of other users but not all.

I guess that the filter should likely start with `User Created / Is one of / ?,?,?`, but how could the list of subordinates be kept and parametrized?

**Comment by rijkvanzanten** [comment]
There's currently no way to configure that type of "subquery" in permissions. Make sure to open [a new feature request](https://github.com/directus/directus/discussions/new?category=feature-requests) detailing how that could work and we might be able to make it happen 🙂

**Reply by luciandavidescu** [reply]
Great! #17491

**Comment by claytongulick** [comment]
We implement similar functionality by maintaining an access table that's a M2M with the entities that should have different access permissions.

For example, our users may need the ability to access/edit items from one or more different clients. For this, we have a client_access M2M with directus_users that has the client_id and the user_id. Then the permissions check is able to look at $CURRENT_USER.client_access.client_id is not null to determine whether they have permission to the item.

We've tested it extensively and have been using it in production without issues for six months now.

**Reply by luciandavidescu** [reply]
Sounds interesting... so there should be subfield access as we speak?

I'm not sure I understand how this is supoosed to work, does it allow generic permissions on multiple other users' items at any point, or does it need a custom query every time? 

What I had tried is to access the m2m with user_created » is one of » $CURRENT_USER.team and it didn't work, nothing at all showed up for that user.

**Reply by claytongulick** [reply]
@luciandavidescu yep, but you have to add one more thing to your query, include the $CURRENT_USER.team.[id field] 

For example, here's one of our permissions that says that our Survey Team Member role is only able to access alerts that are generated for clients that they are assigned to:

```
{
	"_and": [
		{
			"alert_id": {
				"client_id": {
					"id": {
						"_in": [
							"$CURRENT_USER.client_access.client_id"
						]
					}
				}
			}
		}
	]
}
```

**Comment by luciandavidescu** [comment]
I did try `$CURRENT_USER.team.id` as well... 

Just to make sure that we're on the same page, I'm trying to achieve this in an admin only context, so filters should be added like this: 

![filter](https://user-images.githubusercontent.com/26859276/218271826-075be104-7ca5-436a-9eb1-2ff8caae7950.png)

...while admin content lists & permissions should reflect that right away.

**Reply by claytongulick** [reply]
I think I recall seeing in the source code that admin checks bypass the permissions system entirely, so if you have an admin user, I don't think permissions will apply.

**Comment by luciandavidescu** [comment]
@claytongulick and @rijkvanzanten pointed out that this is actually possible:

1. Create a M2M field under directus_users connecting with directus_users ('team' or whatever).

2. Directus will generate a junction table named junction_directus_users_directus_users with 3 columns – id (integer), directus_users_id (current user UUID) and related_directus_users_id (team UUIDs). 

3. In the permissions filter, do [User] » Is one of » $CURRENT_USER.team.related_directus_users_id

Then just add users to an user's 'team' or whatever field and it should work.
